yaml-tests/src/test/java/YamlTestBase.java

```diff
@@ -65,10 +66,11 @@ protected Map<String, Object> getAdditionalOptions() {
     }
 
     protected final void doRun(@Nonnull final String fileName) throws Exception {
-        doRun(fileName, false);
+        doRun(fileName, EnumSet.noneOf(YamlRunner.YamlRunnerOptions.class));
+        // doRun(fileName, EnumSet.of(YamlRunner.YamlRunnerOptions.CORRECT_STATS));
```

Comment:
Normally, I'd suggest removing this commented out line, but maybe what we really need here is documentation on how to run the tests with different options--maybe even pre-baked `EnumSet`s that, theoretically, we could even hook up to certain gradle args, so if you ran `./gradlew :yaml-tests:test -PcorrectStats` or something, it would do this rewriting for you.

I don't think the latter needs to happen now necessarily

Reply:
So I think @ScottDugas  added the ability to connect some options to the YamlTestConfig. I could try to get these other options for correcting the yaml files under that umbrella. Not sure, though, since these other options are somewhat orthogonal, and the options that we hand into to the runner are one-offs you never want to have on when you check in.

Reply:
I added a mechanism to the extension that allows test case maintenance through an annotation:

```
@MaintainYamlTestConfig(YamlTestConfigFilters.CORRECT_EXPLAIN_AND_METRICS)
```

which can be on the test method or the test class in question. Other enum values are allowed to only update the explain or only update the metrics.